scripts/functional-tests.sh

```diff
@@ -8,6 +14,11 @@ export SCRIPTDIR=$(dirname "$0")
 # build a prompt string that includes the time, source file, line number, and function name
 export PS4='+$(date +"%Y-%m-%d %T") ${BASH_VERSION}:${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
 
+# Support overriding the test directory for local testing otherwise creates a temporary directory
+TEST_DIR=${TEST_DIR:-$(mktemp -d)}
+# Support cleaning up the test directory once the script is done, defaults to True, set any value to disable
+export CLEANUP_TEST_DIR
```

Comment:
Actually, the semantics and the variable name don't seem to match. The name would suggest that setting it as `CLEANUP_TEST_DIR=1` would *enable* the cleanup. But the implementation seems reversed: it will *disable* the cleanup when the variable is set. Consider renaming the variable to e.g. `SKIP_CLEANUP`.

---

You could also make the script automatically detect if `TEST_DIR` was set by the user - and disable the cleanup then without a separate flag variable. What's the case to pass `TEST_DIR` explicitly while expecting the directory to be removed at the end of the script? (If that's really needed, the caller can just `rm -rf $TEST_DIR` after the script returns.)

Reply:
> Actually, the semantics and the variable name don't seem to match. The name would suggest that setting it as `CLEANUP_TEST_DIR=1` would _enable_ the cleanup. But the implementation seems reversed: it will _disable_ the cleanup when the variable is set. Consider renaming the variable to e.g. `SKIP_CLEANUP`.

Good point I'll will rename it.

> You could also make the script automatically detect if `TEST_DIR` was set by the user - and disable the cleanup then without a separate flag variable. What's the case to pass `TEST_DIR` explicitly while expecting the directory to be removed at the end of the script? (If that's really needed, the caller can just `rm -rf $TEST_DIR` after the script returns.)

Another good point is that cleanup should always happen to prevent the disk from filling up.
If we skip the cleanup and run the script multiple times, users will likely forget to do some cleanup, also passing a dedicated directory helps not having to download the model again.
I think having both gives us better control.